var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
@davydden The problems I see with `libmkl_rt` are that it seems : 
- it could be driven wrong (you can select the ILP64 interface vs. LP64 whereas the size of your integers is only one in your binary files)
- it's multi-threaded by default

Anyhow this discussion may be worth a new issue on its own...